src/main/java/net/uaznia/lukanus/hudson/plugins/gitparameter/GitParameterDefinition.java

```diff
@@ -290,10 +300,12 @@ public String prettyRevisionInfo(GitClient newgit, Revision r) {
 					return Collections.singletonMap(errMsg, errMsg);
 				}
 
-				FetchCommand fetch = newgit.fetch_().from(remoteURL,
-						repository.getFetchRefSpecs());
-				fetch.execute();
 				if (type.equalsIgnoreCase(PARAMETER_TYPE_REVISION)) {
+					// If we use the GitClient#getRemoteReferences function, we don't
```

Comment:
Part of this is to make sure that we only use ls-remote. I guess I started on this before the other commit to get rid of the workspace was committed. I think I assumed incorrectly that you couldn't get every revision using ls-remote. I put this comment here to indicate why I moved this function inside for just the PARAMETER_TYPE_REVISION. This whole merge request doesn't sound needed anymore, if you've already made everything use ls-remote. Fetching every time had made this plug-in unusable for most of our repos, due to their size.